static/js/runtime-config.js

```diff
@@ -32,15 +32,19 @@ export default class RuntimeConfig {
   set (key, value) {
     this._validateSet(key, value);
     this._data[key] = value;
-    this._onUpdateListener(this.getAll());
+    if (!this._onUpdateListeners[key]) {
+      this._onUpdate(key, () => {});
+    }
+    this._onUpdateListeners[key](value);
   }
 
   /**
-   * A function which is called any time the RuntimeConfig is updated
+   * A function which is called any time a specific key in RuntimeConfig is updated
+   * @param {string} key
    * @param {Function} cb 
    */
-  _onUpdate (cb) {
-    this._onUpdateListener = cb;
+  _onUpdate (key, cb) {
```

Comment:
If we do refactor such that` _onUpdate` stays, the same, I think we should use the function name `registerListener` for registering single listeners so that we are more consistent with storage in Answers: https://github.com/yext/answers-search-ui/blob/master/src/core/storage/storage.js.

Reply:
What do you think about using `_onUpdate (cb, key = null)` for both general and specific key listeners (if key is specified). I could keep `_onUpdate` for the general listener and `registerListener` for the specific one, but would that be confusing or easy to mix up?

Reply:
What do you think about `registerListener({ event: 'update', key: '_linkTarget', cb: () => {} });` for key specific listeners and `registerListener({ event: 'update', cb: () => {} });` for generic update listeners?